controllers/grid/settings/sections/SectionGridHandler.inc.php

```diff
@@ -202,19 +202,32 @@ function updateSection($args, $request) {
 	 */
 	function deleteSection($args, $request) {
 		$journal = $request->getJournal();
-
+		
 		$sectionDao = DAORegistry::getDAO('SectionDAO');
 		$section = $sectionDao->getById(
 			$request->getUserVar('sectionId'),
 			$journal->getId()
 		);
-
+		
 		if ($section && $request->checkCSRF()) {
```

Comment:
We have a specific error message for CSRF failure. It'd be good to use that. Also, the conditionals could be restructured to reduce nesting, but just a nitpick:

```
if (!$request->checkCSRF()) {
	return new JSONMessage(false, __('form.csrfInvalid'));
}

AppLocale::requireComponents(LOCALE_COMPONENT_PKP_MANAGER);
if (!$section) {
	return new JSONMessage(false, __('manager.setup.errorDeletingItem'));
}

if ($checkSubmissions->numRows() > 0) {
	return new JSONMessage(false, __('manager.sections.alertDelete'));
}

$sectionDao->deleteObject($section);
return DAO::getDataChangedEvent($section->getId());
```

Reply:
a lot better